### PR TITLE
More functions that turn integers into floats

### PR DESCRIPTION
**Pull request by antoine-levitt**
Probably not all of them, just the simple ones I could find.

### CODE REVIEW

File: src/ChangePrecision.jl [hunk 1]
```diff
@@ -9,7 +9,8 @@ export @changeprecision
 
 const randfuncs = (:rand, :randn, :randexp) # random-number generators
 const matfuncs = (:ones, :zeros, :eye) # functions to construct arrays
-const intfuncs = (:/, :inv, :sqrt, :√, :float) # functions that turn integers into floats
+const intfuncs = (:/, :inv, :sqrt, :√, :float, :sin, :cos, :tan, :asin, :acos, :acosh, :atanh, :log, :log2, :log10,
+                  :lgamma, :log1p, :cbrt, :sinh, :cosh, :tanh, :atan, :asinh, :exp2, :expm1) # functions that turn integers into floats
```
Comment (stevengj) [inline src/ChangePrecision.jl]: Probably `sind` and `cosd` as well, which might actually be commonly called with integer angles.  And also `exp`.

Comment (stevengj) [reply src/ChangePrecision.jl]: Also `:∛`, which is equivalent to a call to `cbrt`.

File: src/ChangePrecision.jl [hunk 2]
```diff
@@ -9,7 +9,21 @@ export @changeprecision
 
 const randfuncs = (:rand, :randn, :randexp) # random-number generators
 const matfuncs = (:ones, :zeros, :eye) # functions to construct arrays
-const intfuncs = (:/, :inv, :sqrt, :√, :float) # functions that turn integers into floats
+#from https://docs.julialang.org/en/release-0.6/manual/mathematical-operations/, up to date as of 0.6
+const intfuncs = (:/,:inv,:√,:float,:deg2rad,:rad2deg,:cospi,:sinpi,
+                  # powers logs and roots
+                  :sqrt,:cbrt,:hypot,:exp,:expm1,:log,:log2,:log10,:log1p,
```
Comment (stevengj) [inline src/ChangePrecision.jl]: `exp2` and `exp10`

File: src/ChangePrecision.jl (hunk 1 above)
Comment (antoine-levitt) [reply src/ChangePrecision.jl]: Was just committing this :-) good catch on cbrt. To be on the safe side I systematically added everything (that returned floats) in https://docs.julialang.org/en/release-0.6/manual/mathematical-operations/

File: src/ChangePrecision.jl [hunk 3]
```diff
@@ -9,7 +9,21 @@ export @changeprecision
 
 const randfuncs = (:rand, :randn, :randexp) # random-number generators
 const matfuncs = (:ones, :zeros, :eye) # functions to construct arrays
-const intfuncs = (:/, :inv, :sqrt, :√, :float) # functions that turn integers into floats
+#from https://docs.julialang.org/en/release-0.6/manual/mathematical-operations/, up to date as of 0.6
+const intfuncs = (:/,:inv,:√,:∛,:float,:deg2rad,:rad2deg,:cospi,:sinpi,
```
Comment (stevengj) [inline src/ChangePrecision.jl]: also `:\`

File: src/ChangePrecision.jl [hunk 4]
```diff
@@ -9,7 +9,21 @@ export @changeprecision
...
+                  # powers logs and roots
+                  :sqrt,:cbrt,:hypot,:exp,:expm1,:log,:log2,:log10,:log1p,
```
Comment (stevengj) [inline src/ChangePrecision.jl]: `exp2` and `exp10`

Comment (antoine-levitt) [reply src/ChangePrecision.jl]: Those aren't in the manual, will make a PR

**Comment by stevengj** [comment]
Travis failure is unrelated, looks good!

**Pull request merged**